test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/TypePromotionUtilsTests.cs

```diff
@@ -462,6 +462,52 @@ public void PrimitiveTypesOfSameKindCanConvertToEachOther()
             Assert.True(TypePromotionUtils.CanConvertTo(null, primitiveType, stringType));
         }
 
+        [Fact]
+        public void DifferentTypesCanNotConvertToEachOther()
+        {
+            var stringPrimitiveType = new EdmPrimitiveTypeReference(EdmCoreModel.Instance.GetPrimitiveType(EdmPrimitiveTypeKind.String), true);
+            var entityType = new EdmEntityType("NS", "Entity").GetTypeReference();
+            var complexType = new EdmComplexType("NS", "Complex").GetTypeReference();
+            var enumType = new EdmEnumTypeReference(new EdmEnumType("NS", "MyName", false), false);
+
+            Assert.False(TypePromotionUtils.CanConvertTo(null, stringPrimitiveType, complexType));
+            Assert.False(TypePromotionUtils.CanConvertTo(null, stringPrimitiveType, entityType));
+			// String to enum is allowed!?
+			//Assert.False(TypePromotionUtils.CanConvertTo(null, stringPrimitiveType, enumType));

```

Comment:
Yes; in OData 4.01 we added support for implicit conversions between string and enum. Since it was non-breaking, we made it available for 4.0 as well. We can remove the commented line.